src/test/java/itest/TargetRecordingPatchIT.java

```diff
@@ -0,0 +1,178 @@
+ * licensable by each licensor hereunder covering either (i) the unmodified
+ * Software as contributed to or provided by such licensor, or (ii) the Larger
+ * Works (as defined below), to deal in both
+ *
+ * (a) the Software, and
+ * (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
+ * one is included with the Software (each a "Larger Work" to which the Software
+ * is contributed by such licensors),
+ *
+ * without restriction, including without limitation the rights to copy, create
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package itest;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import io.vertx.core.MultiMap;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import itest.bases.StandardSelfTest;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class TargetRecordingPatchIT extends StandardSelfTest {
+    static final String TEST_RECORDING_NAME = "someRecording";
+    static final String RECORDING_REQ_URL =
+            String.format("/api/v1/targets/%s/recordings", SELF_REFERENCE_TARGET_ID);
+    static final String ARCHIVED_REQ_URL = "/api/v1/recordings";
+    static final String OPTIONS_REQ_URL =
+            String.format("/api/v1/targets/%s/recordingOptions", SELF_REFERENCE_TARGET_ID);
+
+    @Test
+    void testSaveEmptyRecordingDoesNotArchiveRecordingFile() throws Exception {
+        try {
+            // Limit recording size to 0 bytes to guarantee an empty recording
+            CompletableFuture<JsonObject> dumpResponse = new CompletableFuture<>();
+            MultiMap optionsForm = MultiMap.caseInsensitiveMultiMap();
+            optionsForm.add("maxAge", "0");
+            optionsForm.add("toDisk", "false");
+            optionsForm.add("maxSize", "0");
+
+            webClient
+                    .patch(OPTIONS_REQ_URL)
```

Comment:
These options can be set on a per-recording basis too:

https://github.com/cryostatio/cryostat/blob/main/src/main/java/io/cryostat/net/web/http/api/v1/TargetRecordingsPostHandler.java#L173

It would be more concise and less prone to causing external test breakage if these settings were set for the new recording, rather than being set as the defaults for the target.

Reply:
If I post a recording with `maxSize: 1` or  `maxSize: 0`, there seems to be data returned when saving the recording. Only if I set the recording options as target defaults am I able to receive a `204` from the save handler.

```
...
MultiMap form = MultiMap.caseInsensitiveMultiMap();
            form.add("recordingName", TEST_RECORDING_NAME);
            form.add("duration", "5");
            form.add("events", "template=ALL");
            form.add("maxSize", "1");

            webClient
                    .post(RECORDING_REQ_URL)
...

java.lang.AssertionError: 
Expected: <204>
     but: was <200>
```

Reply:
Hmm. I've been looking over the `TargetRecordingOptionsPatchHandler` again and how it works (`RecordingOptionsCustomizer`, `RecordingOptionsBuilderFactory`) and comparing that to the `TargetRecordingsPostHandler`, and I'm not sure why there would be this difference in behaviour. This might be a separate bug to investigate. For now just leave the request as you have it, then.